pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp

```diff
@@ -692,6 +692,177 @@ class StateVectorCudaManaged
         }
         return expval(obsName, wires, params, matrix_cu);
     }
+    /**
+     * @brief See `expval(std::vector<CFP_t> &gate_matrix = {})`
+     */
+    auto expval(const std::vector<size_t> &wires,
+                const std::vector<std::complex<Precision>> &gate_matrix) {
+
+        std::vector<CFP_t> matrix_cu(gate_matrix.size());
+
+        for (std::size_t i = 0; i < gate_matrix.size(); i++) {
+            matrix_cu[i] =
+                cuUtil::complexToCu<std::complex<Precision>>(gate_matrix[i]);
+        }
+
+        if (gate_matrix.empty()) {
+            std::string message = "Currently unsupported observable";
+            throw LightningException(message.c_str());
+        }
+
+        auto &&local_wires =
```

Comment:
Why do we reverse wires when the gate_matrix is not empty?

Reply:
cuQuantum expects the matrix in a given order. The quickest way here to get this was to reverse the wire indices, rather than touching the matrix itself.

Reply:
How about commenting it in the source code?

Reply:
Good call. This was commented in the other overloaded `expval` but not here. I'll add a mention.